controllers/medusa/medusabackupschedule_controller.go

```diff
@@ -0,0 +1,142 @@
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	medusav1alpha1 "github.com/k8ssandra/k8ssandra-operator/apis/medusa/v1alpha1"
+	cron "github.com/robfig/cron/v3"
+)
+
+type Clock interface {
+	Now() time.Time
+}
+
+// MedusaBackupScheduleReconciler reconciles a MedusaBackupSchedule object
+type MedusaBackupScheduleReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Clock  Clock
+}
+
+type RealClock struct{}
+
+func (r *RealClock) Now() time.Time {
+	return time.Now()
+}
+
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules/finalizers,verbs=update
+// +kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=cassandrabackups,verbs=get;list;watch;create
+
+func (r *MedusaBackupScheduleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithValues("medusabackupscheduler", req.NamespacedName)
+
+	backupSchedule := &medusav1alpha1.MedusaBackupSchedule{}
+	err := r.Get(ctx, req.NamespacedName, backupSchedule)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	sched, err := cron.ParseStandard(backupSchedule.Spec.CronSchedule)
+	if err != nil {
+		// The schedule is in incorrect format
+		return ctrl.Result{}, err
+	}
+
+	previousExecution, err := getPreviousExecutionTime(ctx, backupSchedule)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	now := r.Clock.Now().UTC()
+
+	// Calculate the next execution time
+	nextExecution := sched.Next(previousExecution).UTC()
+
+	createBackup := false
+
+	if nextExecution.Before(now) {
+		nextExecution = sched.Next(now)
+		previousExecution = now
+		createBackup = true && !backupSchedule.Spec.Disabled
+	}
+
+	// Update the status if there are modifications
+	if backupSchedule.Status.LastExecution.Time.Before(previousExecution) ||
+		backupSchedule.Status.NextSchedule.Time.Before(nextExecution) {
+		backupSchedule.Status.NextSchedule = metav1.NewTime(nextExecution)
+		backupSchedule.Status.LastExecution = metav1.NewTime(previousExecution)
+
+		if err := r.Client.Status().Update(ctx, backupSchedule); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	if createBackup {
+		// TODO Verify here no previous jobs is still executing?
+		logger.V(1).Info("Scheduled time has been reached, creating a backup job", "MedusaBackupSchedule", req.NamespacedName)
+		generatedName := fmt.Sprintf("%s-%d", backupSchedule.Name, now.Unix())
+		backupJob := &medusav1alpha1.MedusaBackupJob{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generatedName,
+				Namespace: backupSchedule.Namespace,
+				// TODO Labels?
```

Comment:
Should this inherit the labels from the backup schedule, or from the cassandradatacenter?

Reply:
I opened https://github.com/k8ssandra/k8ssandra-operator/issues/643.